include/seqan3/std/ranges

```diff
@@ -173,8 +179,171 @@ template <typename type>
+ */
+using ::ranges::copy;
+
+/*!\interface std::ranges::empty
+ * \brief Alias for ranges::empty. Checks whether a range is empty.
+ */
+using ::ranges::empty;
+
+/*!\interface std::ranges::equal
+* \brief Alias for ranges::equal. Determines if two sets of elements are the same.
+*/
+using ::ranges::equal;
+
+/*!\interface std::ranges::default_sentinel
+* \brief Alias for ranges::default_sentinel. Empty sentinel type for use with iterator types that know the bound of their range.
+*/
+using ::ranges::default_sentinel;
+
+/*!\interface std::ranges::sentinel_t
+* \brief Alias for ranges::sentinel_t. Obtains the sentinel type of a range.
+*/
+using ::ranges::sentinel_t;
+
+/*!\interface std::ranges::reference_t
+* \brief Alias for ranges::reference_t. Obtains a dereferenceable object's reference type.
+*/
+using ::ranges::reference_t;
+
+/*!\interface std::ranges::value_type_t
+* \brief Alias for ranges::value_type_t. Obtains the value type of a Readable type.
+*/
+using ::ranges::value_type_t;
+
+/*!\interface std::ranges::rvalue_reference_t
+* \brief Alias for ranges::rvalue_reference_t. Obtains a dereferenceable object's rvalue reference type.
+*/
+using ::ranges::rvalue_reference_t;
+
+/*!\interface std::ranges::difference_type_t
+* \brief Alias for ranges::difference_type_t. Obtains the difference type of a WeaklyIncrementable type.
+*/
+using ::ranges::difference_type_t;
+
+/*!\interface std::ranges::size_type_t
+* \brief Alias for ranges::size_type_t. Obtains the value type of a Size type.
+*/
+using ::ranges::size_type_t;
+
+/*!\interface std::ranges::iterator_t
+* \brief Alias for ranges::iterator_t. Obtains the iterator type of a range.
+*/
+using ::ranges::iterator_t;
 
+/*!\interface std::ranges::semiregular_t
+* \brief Alias for ranges::semiregular_t. Obtains the value type of a Semiregular type.
+*/
+using ::ranges::semiregular_t;
+
+/*!\interface std::ranges::common_reference_t
+* \brief Alias for ranges::common_reference_t. Determine the common reference type of a set of types.
+*/
+using ::ranges::common_reference_t;
+
+/*!\interface std::ranges::common_type_t
+* \brief Alias for ranges::common_type_t. Determine the common type of a set of types.
+*/
+using ::ranges::common_type_t;
+
+/*!\interface std::ranges::uncvref_t
+* \brief Alias for ranges::uncvref_t. Obtains the reference type of a top level cv.
+*/
+using ::ranges::uncvref_t;
+
+/*!\interface std::ranges::range_value_type_t
+* \brief Alias for ranges::range_value_type_t. Obtains the value type of a Range type.
+*/
+using ::ranges::range_value_type_t;
+
+/*!\interface std::ranges::ostreambuf_iterator
+* \brief Alias for ranges::ostreambuf_iterator. Output iterator that writes to std::basic_streambuf.
+*/
+using ::ranges::ostreambuf_iterator;
+
+/*!\interface std::ranges::back_insert_iterator
+* \brief Alias for ranges::back_insert_iterator. Iterator adaptor for insertion at the end of a container.
+*/
+using ::ranges::back_insert_iterator;
```

Comment:
We need to think about something here, because the above two should not be in the `std/ranges`, but in `std/iterator`, but then they will conflict with the standard library one :(

Reply:
So std::ranges::back_insert_iterator is no longer used, but std::ranges::back_inserter is still used, correct? Do we have a solution for the fact that it should be in std/iterator?